### PR TITLE
native: add -t <port> option to native again

### PR DESCRIPTION
**Pull request by cgundogan**
This PR brings back the option to set a port and daemonize native in order to use it with desvirt or netcat.
Can be tested with  `./bin/native/gnrc_networking -t 12107` and then `nc localhost 12107`.

@emmanuelsearch @thomaseichinger can anyone of you test this on OS X?

### CODE REVIEW

**Comment by kaspar030** [comment]
This shouldn't be part of riot.

**Comment by OlegHahm** [comment]
> This shouldn't be part of riot.

What and why not?

**Comment by kaspar030** [comment]
Sorry, let me rephrase. IMHO this should not be in the RIOT native port, but be provided by a small external tool.

**Comment by authmillenon** [comment]
So instead of putting stdio through something else than stdio (sockets in this case) we should parse stdio and provide UART-like behavior through an external tool?!?

**Comment by kaspar030** [comment]
No. RIOT native already has in-RIOT stdio <-> linux stdio. Adding support to pipe all that through a socket can of course be done within RIOT, but it adds a lot of unnecessary complexity. I'd rather see a small tool that does the job from the outside.

Maybe we can even reuse something existing, e.g., ```socat EXEC:bin/native/default.elf,end-close,stderr,pty TCP-L:8888,reuseaddr,fork```.

**Comment by cgundogan** [comment]
+1 for the socat idea. Maybe it's really the better route to go with external tools here?

**Comment by OlegHahm** [comment]
:+1:

**Comment by cgundogan** [comment]
I just tested quickly with socat and one thing I noticed is that the pipe breaks when I attach/detach several times from that running instance. We need to cleverly bypass this somehow

**Comment by kaspar030** [comment]
@cgundogan How do you test that?

**Comment by cgundogan** [comment]
@kaspar030 can oyu have a look at https://github.com/des-testbed/desvirt/pull/17 ? Is this what you had in mind?

**Comment by LudwigOrtmann** [comment]
I have no particular opinion on the topic.

**Comment by kaspar030** [comment]
I'll close this in favour of https://github.com/des-testbed/desvirt/issues/17.
